packages/eslint-plugin/src/rules/no-shadow.ts

```diff
@@ -564,6 +575,11 @@ export default createRule<Options, MessageIds>({
           continue;
         }
 
+        // ignore variables of function declarations or function/method overloads
+        if (isParameterOfTypeOnlyFunctionDeclaration(variable)) {
```

Comment:
i'm wondering if these added checks should be part of `isFunctionTypeParameterNameValueShadow`, so that they're affected by the `ignoreFunctionTypeParameterNameValueShadow` option? I was playing with this playground link, and the inconsistencies are the same but inverted compared to the issue report: https://deploy-preview-10543--typescript-eslint.netlify.app/play/#ts=5.7.2&fileType=.ts&code=CYUwxgNghgTiAEkoGdnwMoHsC2IDC0qAjPAN4BQ8imAdsgC4wCuY9mMAFGLQ86%2BwAVYUbAC54UGgE8AlAG5K8ZDhABZEPQAWmYB1xadQmCPGTZCqstzqDwAOoBLLQFEAHgAcIDsE4BKGphgaABUpdxA9DW1gIxEAfVNpGXFeBxoAcwt4AHps%2BAApJgYJeAAzEAB3eGx2BHoQBmQAOhbFKzUonQBBMDAQd3o09LwoCAgAIygwAGsuUYmp6fFI21ixCST4AF4APg3zNpUBGExwmHopZdgTioAxJhpWB1o1xNltvYA3TAdgBQBfcilB5PWjUOiMFhsGBrDgyMjwQHAx6DMH6aKw%2BGkRFAkGomjVToxYTYOJwhFIvHPAnXTB3KkvEnk7GA8i5eAtJoYEQITRQdzhGhDeAVJyaeBaBAicYOEA0ehlBkE0BIYz40Tkcgq6BwRUo6lKFT3fW0DjI0E0V77ZLwb6-BTm-F6i2YikKNl5TnwcZMBU0TAK0VaCWaKXYGVyhWfWAOKDjCAIbXCJ0caMwbIJ%2BjZbgQ%2BAgVwiTwNZKapO6tOG3AANRjcYTyzA80mMytZnhu1tPz%2BuJNBMbY2b01dLLkOTyXQgyng-rzMBOMDQDlKttgGY02Z4CocaCKIGA8DSDBAUH3obgpfAOoQmcrIBrMFj8ZAABkNA2m4s1q-6G9218uw6SqIB%2BLYkt%2BzKIu6ZYIDmxTtPej4Jngm7vgOn4kshEK-h8nb2j2FrAWhoHGNgmEMBB-zuuyABymCzvOi7LpKSh8sAdJ7tOPIHmgAAGaY8RqWqXrAMGEGgWC4AQKDIAATGQiiwZC-CcIpfDQmsMm-lk7Q2NEKwYiSmnWgC5AVqpUKCIZCgVuihhWReqoII6BrtMaFpmkqGk2na3YVs5jIkTJVGei08DsPAPGZgJDlXogYkYCoUmoAAzPJVDmcpXCbmplkkclWmHNYRL6XZeUFYCN4ZepJLJQoN62cSeVQcJur%2BQSrlKh5vZrMl3kAaZsDOviPXBRyoXhTxinRUJjlxdJCWSWJAAsaXgrwFkqdlG1rEtBWWCouk6CVjUiLtxnkICilrUp1UkStWzwOSHYjuQV0NTt2yPX%2BboxSJQ0uUanVtTtfV4X5nkkktChAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaZACwEMATfAd3SnwDMHIAacbSAAQBcBPABxQBjaAEt%2B3Qinijc3APQFi5KrQwBtdtiiJo0fNFZbsWbR1EBzAtEQAxWLiHdR%2BXABUBiAAploZALaI3LoAcgGIAGpk8LCIAMoqNHQM0ciIxhAAvsYAuuzZmUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

Reply:
Oh, I've completely missed the mark on this 🙈 

I initially considered it a false positive and thought the parameter's value could never be referenced directly. Looking some more into it, and especially after reading https://github.com/typescript-eslint/typescript-eslint/issues/2447#issuecomment-684021953, I see it is possible, and a lot like other checks behind the `isFunctionTypeParameterNameValueShadow` option.

I've updated the PR to match this, thanks for catching this 👍